rad/repl.rad

```diff
@@ -11,15 +11,21 @@
 
   Type (doc '<name>) for further documentation of <name>."))
 
-(def read-line-or-exit! (fn []
-  (def line (get-line!))
-  (if (eq? line nil)
-      (exit!)
-      (read line))))
-
-(def repl
-  {:name "repl"
-   ;; Parse errors are thrown here
-   :getter read-line-or-exit!})
-
-(subscribe-to! repl print!)
+(def render-exception! (fn [e stack]
+  (put-str! (string-append "Exception: " (show e)))
+  (for-each-vec stack put-str!)))
+
+(def repl (fn []
+  (def-rec repl-rec (fn [st]
+    (def line (get-expression! (keys (lookup :env st))))
+    (def cur-eval (nth 1 (lookup 'eval (lookup :env st))))
+    (if (eq? line nil)
+        nil
+        (catch 'any (do (def ret (cur-eval (read line) st))
```

Comment:
This makes parse errors print a lot more poorly.

Reply:
Fixed.